src/services/teams/hooks/helpers.js

```diff
@@ -0,0 +1,481 @@
+                return false;
+
+            let bool = false;
+            oldTeam.invitedUserIds.map(inv => {
+                if (isDefined(inv.email) && 
+                    ['teamexpert','teamadministrator'].includes(inv.role) && 
+                    addingUser.email === inv.email && 
+                    inv.role===hook.findRole('_id',addingTeamUser.role,'name')) {
+                        bool = true;
+                }
+            });
+            return bool;
+
+        } else {
+            return true;
+        }
+    }
+    return false;
+};
+
+/**
+* @helper
+* @requires function::extractOne
+* @param {Object::hook} hook
+* @return {Promise::Object::team}
+*/
+const getTeam = (hook) => {
+    return new Promise((resolve, reject) => {
+        const method = hook.method;
+        const teamId = hook.id || (hook.result || {})._id || hook.teamId || get(hook, 'teamId');
+        const sessionUserId = bsonIdToString(hook.params.account.userId);
+        const restrictedMatch = { $elemMatch: { userId: sessionUserId } };
+        const teamsService = hook.app.service('teams');
+
+        const resolveIt = data => {
+            set(hook, 'team', data);    //set it for later use 
+            resolve(data);
+        };
+
+        if (isDefined(get(hook, 'team'))) {
+            resolveIt(get(hook, 'team'));
+        } else if (method === 'create') {
+            resolveIt(hook.data);
+        } else if (teamId) {
+            const query = isAcceptWay(hook, teamId) ? { _id: teamId } : { _id: teamId, userIds: restrictedMatch };
+            teamsService.find({ query }).then(_teams => {
+                extractOne(_teams).then(_team => {
+                    resolveIt(_team);
+                }).catch(err => {
+                    reject(err);
+                });
+            }).catch(err => {
+                reject(new errors.NotFound('Can not found team with teamId=' + teamId, err));
+            });
+        } else if (method === 'find') {
+            teamsService.find({
+                query: { userIds: restrictedMatch }
+            }).then(_teams => {
+                resolveIt(_teams.data);
+            }).catch(err => {
+                reject(new errors.NotFound('Can not found team for user with userId=' + sessionUserId, err));
+            });
+        } else {
+            throw new errors.NotImplemented('It should not run into this case.');
+        }
+    }).catch(err => {
+        logger.warn(err);
+        throw new errors.NotFound('Can not found this team.');
+    });
+};
+
+
+
+/**
+*   @helper
+*   @private
+*   @requires function::hook.findRole Can added over hook function ->todo name
+*   @param {Object::hook} hook
+*   @param {String} userId
+*   @param {String} selectedRole
+*/
+const createUserWithRole = (hook, userId, selectedRole) => {
+    let role;
+    if (isUndefined(selectedRole))
+        role = hook.findRole('name', 'teammember', '_id'); //roles.teammember;
```

Comment:
Unnecessary comment

Reply:
It is important if the role in undefinied.